twoliter/src/lock.rs

```diff
@@ -252,48 +258,42 @@ impl OCIArchive {
 }
 
 /// Represents the structure of a `Twoliter.lock` lock file.
-#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Serialize, Deserialize)]
 #[serde(rename_all = "kebab-case")]
 pub(crate) struct Lock {
     /// The version of the Twoliter.toml this was generated from
     pub schema_version: SchemaVersion<1>,
-    /// The workspace release version
-    pub release_version: String,
     /// The resolved bottlerocket sdk
     pub sdk: LockedImage,
     /// Resolved kit dependencies
     pub kit: Vec<LockedImage>,
-    /// sha256 digest of the Project this was generated from
-    pub digest: String,
 }
 
 #[allow(dead_code)]
 impl Lock {
-    pub(crate) async fn load(project: &Project) -> Result<Self> {
-        let lock_file_path = project.project_dir().join(TWOLITER_LOCK);
-        if lock_file_path.exists() {
-            let lock_str = read_to_string(&lock_file_path)
-                .await
-                .context("failed to read lockfile")?;
-            let lock: Self =
-                toml::from_str(lock_str.as_str()).context("failed to deserialize lockfile")?;
-            // The digests must match, if changes are needed twoliter
-            ensure!(lock.digest == project.digest()?, "changes have occurred to Twoliter.toml that require an update to Twoliter.lock, if intentional please run twoliter update");
-            return Ok(lock);
-        }
-        Self::create(project).await
-    }
-
     pub(crate) async fn create(project: &Project) -> Result<Self> {
         let lock_file_path = project.project_dir().join(TWOLITER_LOCK);
-        if lock_file_path.exists() {
-            remove_file(&lock_file_path).await?;
-        }
-        let lock = Self::resolve(project).await?;
-        let lock_str = toml::to_string(&lock).context("failed to serialize lock file")?;
+        let lock_state = Self::resolve(project).await?;
+        let lock_str = toml::to_string(&lock_state).context("failed to serialize lock file")?;
         write(&lock_file_path, lock_str)
             .await
             .context("failed to write lock file")?;
+        Ok(lock_state)
+    }
+
+    pub(crate) async fn load(project: &Project) -> Result<Self> {
+        let lock_file_path = project.project_dir().join(TWOLITER_LOCK);
+        let lock_state = Self::resolve(project).await?;
+        ensure!(
+            lock_file_path.exists(),
+            "Twoliter.lock does not exist please run twoliter update first"
+        );
+        let lock_str = read_to_string(&lock_file_path)
```

Comment:
nit: You could move the resolution below the `ensure` to save some work in the case that the workspace isn't created yet.